torch/csrc/jit/codegen/cuda/lower_utils.cpp

```diff
@@ -277,6 +337,195 @@ std::pair<kir::ForLoop*, int64_t> getAllocPoint(
 }
 
 } // namespace loop_utils
+
+namespace kir_utils {
+
+namespace {
+
+class ReplaceExprInput : public kir::MutableIrVisitor {
+ public:
+  static kir::Expr* replace(
+      kir::Expr* expr,
+      const std::unordered_map<kir::Val*, kir::Val*>& replacement_map) {
+    ReplaceExprInput replacer(expr, replacement_map);
+    expr->accept(&replacer);
+    TORCH_INTERNAL_ASSERT(expr != nullptr);
+    TORCH_INTERNAL_ASSERT(replacer.replaced_expr_ != nullptr);
+    auto ret_expr = replacer.replaced_expr_;
+
+    // Copy predicates if the original expr is predicated
+    if (ret_expr != expr) {
+      ret_expr->setPredicate(expr->predicate());
```

Comment:
In #1051, I added `write_predicate_`, which should be copied as well. If this PR is merged first, I'll address this in #1051.

Reply:
Thanks for letting me know. I will look into expanding this one.

Reply:
I just merged 1051

Reply:
Already merged TOT and updated this PR to copy the `write_predicate` so this common utility pass is valid.